core/src/main/java/org/apache/brooklyn/util/core/config/ConfigBag.java

```diff
@@ -126,7 +127,11 @@ public void markUsed(String key) {
     /** As {@link #newInstanceExtending(ConfigBag)} but also putting the supplied values. */
     @Beta
     public static ConfigBag newInstanceExtending(final ConfigBag configBag, Map<?,?> optionalAdditionalValues) {
-        return newInstanceExtending(configBag).putAll(optionalAdditionalValues);
+        if (optionalAdditionalValues != null) {
+            return newInstance(CollectionHelpers.mergeNestedMaps(configBag.getAllConfig(), (Map<String, Object>) optionalAdditionalValues));
+        } else {
+            return newInstanceExtending(configBag);
+        }
```

Comment:
I think we shouldn't do this, too intrusive. Better make the merging behaviour specific to `provisioning.properties`.